### PR TITLE
New client attribute API

### PR DESCRIPTION
**Pull request by konraddysput**
# Why

Our Backtrace client doesn't have any programmatic interface to add attributes. We allow our customers to use direct attributes map, however, we don't know when it's happening and because of this we cannot update native integration attributes. All attributes added dynamically by the user during the application lifecycle should be visible in the native integration.

This pull request adds a new API that allows adding an attribute key pair/ multiple attributes to the Backtrace Client and updates managed/native integration attributes 

ref: BT-2255

### CODE REVIEW

File: backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java [hunk 1]
```diff
@@ -296,6 +296,35 @@ public Map<String, Object> getAttributes() {
...
+    }
+    /**
+     * Adds new attributes to Backtrace Client. If the native integration is enabled, adds attributes
+     * to the native report attributes if:
+     *  - the value exists (is not a null)
+     *  - is not an object (the attribute value is primitive type like String, or Int)
+     * @param attributes Map of attributes
+     */
+    public void addAttribute(Map<String, Object> attributes) {
+        for (Map.Entry<String, Object> entry : attributes.entrySet()) {
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java:323]: if attributes will be null it will crash

File: backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java [hunk 2]
```diff
@@ -296,6 +296,35 @@ public Map<String, Object> getAttributes() {
...
+    /**
+     * Adds a new attribute to Backtrace Client. If the native integration is enabled, adds the attribute
+     * to the native report attributes if:
+     *  - the value exists (is not a null)
+     *  - is not an object (the attribute value is primitive type like String, or Int)
+     * @param key attribute name
+     * @param value attribute value.
+     */
+    public void addAttribute(String key, Object value) {
+        attributes.put(key, value);
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java:308]: what if value will be null

File: backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java [hunk 3]
```diff
@@ -50,6 +51,7 @@ public class BacktraceDatabase implements Database {
     private boolean _enable = false;
     private Breadcrumbs breadcrumbs;
 
+    private boolean _enabledNativeIntegration = false;
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java:54]: name can be without _

File: backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java [hunk 4]
```diff
@@ -167,7 +169,7 @@ public Boolean setupNativeIntegration(BacktraceBase client, BacktraceCredentials
     public Boolean setupNativeIntegration(BacktraceBase client, BacktraceCredentials credentials,
                                           boolean enableClientSideUnwinding, UnwindingMode unwindingMode) {
         // avoid initialization when database doesn't exist
-        if (getSettings() == null) {
+        if (_enable == false || getSettings() == null) {
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java:172]: `if (!_enable || getSettings() == null) {`

File: backtrace-library/src/androidTest/java/backtraceio/library/database/BacktraceDatabaseTest.java [hunk 5]
```diff
@@ -52,6 +52,10 @@ public void after() {
         this.database.clear();
     }
 
+    @Test
+    public void shouldNotCrashWhenNativeIntegrationIsNotEnabled() {
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/androidTest/java/backtraceio/library/database/BacktraceDatabaseTest.java:56]: please add test which adding null as key and null as value and check behaviour

File: backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java (hunk 3 above)
Comment (konraddysput) [reply backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java:54]: In following the pattern that we already have there

File: backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java (hunk 2 above)
Comment (konraddysput) [reply backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java:308]: It's fine if the value is null. We want to store this information. We cannot pass it to the native

File: backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java (hunk 4 above)
Comment (konraddysput) [reply backtrace-library/src/main/java/backtraceio/library/BacktraceDatabase.java:172]: I will stick to the current implementation

**Review by BartoszLitwiniuk — APPROVED** [review]
lgtm
